File: other/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int main(void)
{
    string paragraph = get_string("Text: ");
    int num_lett = 0;
    int num_word = 1;
    int num_sen = 0;
    int trail_space = 0;

    for (int i = strlen(paragraph) - 1; paragraph[i] == ' '; i--)
    {
        trail_space++;
    }

    for (int i = 0, n = strlen(paragraph); i < n - trail_space; i++)
    {
        if ((paragraph[i] >= 'a' && paragraph[i] <= 'z') || (paragraph[i] >= 'A' && paragraph[i] <= 'Z'))
        {
            num_lett++;
        }
        else if (paragraph[i] == ' ')
        {
            num_word++;
        }
        else if (paragraph[i] == '.' || paragraph[i] == '!' || paragraph[i] == '\?')
        {
            num_sen++;
        }
    }

    int grade = (int) (5.88 * ((float) num_lett / (float) num_word) - 29.6 * ((float) num_sen / (float) num_word) - 15.3);


    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability/readability.py

paragraph = input("Text: ")
num_lett = 0
num_word = 1
num_sen = 0

for i in range(0, len(paragraph)):
    if (ord(paragraph[i]) >= ord("a") and ord(paragraph[i]) <= ord("z")) or (ord(paragraph[i]) >= ord("A") and ord(paragraph[i]) <= ord("Z")):
        num_lett += 1
    elif paragraph[i] == " ":
        num_word += 1
    elif paragraph[i] == "." or paragraph[i] == "!" or paragraph[i] == "?":
        num_sen += 1

grade = 5.88 * (num_lett / num_word) - 29.6 * (num_sen / num_word) - 15.3


if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {int(grade)}")